src/TemplateProcessor.ts

```diff
@@ -121,7 +121,8 @@ export default class TemplateProcessor {
 
     private tempVars:JsonPointerString[];
 
-
+    /** cleanup procedures which will be called every time a new template is initialized */
```

Comment:
maybe add "Allows caller to set a callback to propagate initialization into their framework"